### PR TITLE
[TIMOB-20147] iOS: Ti.Contacts.showContacts() selectedPerson returns …

### PR DESCRIPTION
**Pull request by sriks**
JIRA https://jira.appcelerator.org/browse/TIMOB-20147

### CODE REVIEW

File: iphone/Classes/ContactsModule.m [hunk 1]
```diff
@@ -899,14 +899,23 @@ -(void)peoplePickerNavigationControllerDidCancel:(ABPeoplePickerNavigationContro
...
--(BOOL)peoplePickerNavigationController:(ABPeoplePickerNavigationController *)peoplePicker shouldContinueAfterSelectingPerson:(ABRecordRef)person
+-(BOOL)peoplePickerNavigationController:(ABPeoplePickerNavigationController *)peoplePicker shouldContinueAfterSelectingPerson:(ABRecordRef)selectedPerson
 {
 	if (selectedPersonCallback) {
-		ABRecordID id_ = ABRecordGetRecordID(person);
-		TiContactsPerson* person = [[[TiContactsPerson alloc] _initWithPageContext:[self executionContext] recordId:id_ module:self] autorelease];
+		TiContactsPerson* person = nil;
+		if ([TiUtils isIOS8OrGreater] && (ABAddressBookGetAuthorizationStatus() != kABAuthorizationStatusAuthorized)) {
+			// In iOS 8 selected contact is returned without requiring user permission. But we cannot query metadata like recordid.
+			person = [[[TiContactsPerson alloc] _initWithPageContext:[self executionContext]
```
Comment (hansemannn) [inline iphone/Classes/ContactsModule.m]: Is this shifted again or is Github messing stuff up? (http://abload.de/img/bildschirmfoto2015-12fna51.png)

Comment (sriks) [inline iphone/Classes/ContactsModule.m]: Looks like objective-c style method formatting is the issue. Changed it to single line. Please check.

File: apidoc/Titanium/Contacts/Person.yml [hunk 2]
```diff
@@ -264,7 +264,9 @@ properties:
     
   - name: recordId
     summary: Record identifier of the person. Single value. Deprecated since iOS 9.
-    description: For iOS 9 and later, use the [identifier](Titanium.Contacts.Person.identifier) property.
+    description: For iOS 9 and later, use the [identifier](Titanium.Contacts.Person.identifier) property. 
+                 In iOS 8, a contact can be selected without requiring User permission. In such case this 
+                 property returns -1.
```
Comment (hansemannn) [inline apidoc/Titanium/Contacts/Person.yml]: Does the ``validate.py`` accept this change? The format for multiline yaml should be:

```yaml
description: |
    For iOS 9 and later, use the [identifier](Titanium.Contacts.Person.identifier) property. 
    In iOS 8, a contact can be selected without requiring User permission. In such case this 
    property returns -1.
```

Comment (sriks) [inline apidoc/Titanium/Contacts/Person.yml]: validate.py is not complaining. But I did change to use new line format. Please check.

**Pull request merged**